packages/preferences-api/src/browser/preference-service.ts

```diff
@@ -112,7 +112,7 @@ export class PreferenceServiceImpl implements PreferenceService, FrontendApplica
         }
         // tslint:disable-next-line:forin
         for (const preferenceName in preferenceChanges) {
-            this.onPreferenceChangedEmitter.fire(preferenceChanges[preferenceName]);
+            this.onPreferenceChangedEmitter.fire(Object.freeze(preferenceChanges[preferenceName]));
```

Comment:
you should freeze values when they are created and deep, not on access

Reply:
I guess I can freeze them at the beginning of createPreferenceProxy, otherwise to do it before it's going to be the job of the preference contributors.

Reply:
why not here: https://github.com/theia-ide/theia/pull/1175/files#diff-f719560f0cbb68164ea8381e5c8ddd3fR91?

Reply:
Oh yes for the actual preferences that's okay, I was talking more about configs/schemas.